spec.md

```diff
@@ -307,6 +310,22 @@ These limits apply for messages generated by both COs and plugins.
 | 128 bytes  | string              |
 | 4 KiB      | map<string, string> |
 
+### Error Scheme
+
+All CSI API calls defined in this spec MUST return a [standard gRPC status](https://github.com/grpc/grpc/blob/master/src/proto/grpc/status/status.proto).
+Most gRPC libraries provide helper methods to set and read the status fields.
+
+The status `code` MUST contain a [canonical error code](https://github.com/grpc/grpc-go/blob/master/codes/codes.go). COs must handle all valid error codes. Each RPC defines a set of gRPC error codes that MUST be returned by the plugin when specified conditions are encountered. In addition to those, if the conditions defined below are encountered, the plugin MUST return the associated gRPC error code.
+
+Condition | gRPC Code | Description | Recovery Behavior
+| --- | --- | --- | --- |
+| Unsupported request version | 3 INVALID_ARGUMENT | Indicates that the version specified in the request is not supported by the Plugin. | Caller MUST NOT retry; caller SHOULD call `GetSupportedVersions` to discover which CSI versions the Plugin supports. |
```

Comment:
Hmm. This **is** tricky. As @jieyu points out, the [gRPC error code docs](https://github.com/grpc/grpc-go/blob/master/codes/codes.go) state:

> // Note that this differs from FailedPrecondition. It indicates arguments
> // that are problematic regardless of the state of the system

However, the `FailedPrecondition` doc says:

> // FailedPrecondition indicates operation was rejected because the
> // system is not in a state required for the operation's execution.

I read this as a `FailedPrecondition` being an argument's relationship to the state of the system. The question becomes -- are the versions supported by a CSI SP part of the system's state? I don't know. I could go either way here.

Reply:
I'm going to leave it as `InvalidArgument` because I don't want the caller to retry without human intervention.